server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -473,6 +471,8 @@ protected Node(
             ).collect(Collectors.toSet());
             final TransportService transportService = newTransportService(settings, transport, threadPool,
                 networkModule.getTransportInterceptor(), localNodeFactory, settingsModule.getClusterSettings(), taskHeaders);
+            final GatewayMetaState gatewayMetaState = new Zen1GatewayMetaState(settings, nodeEnvironment, metaStateService,
```

Comment:
if we want to flexibly run with Zen1 or Zen2, we need to make this somewhat configurable. This can be handled in a follow-up, however.